agentscope-extensions/agentscope-extensions-scheduler/agentscope-extensions-scheduler-quartz/src/main/java/io/agentscope/extensions/scheduler/quartz/AgentQuartzJob.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2024-2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.agentscope.extensions.scheduler.quartz;
+
+import io.agentscope.extensions.scheduler.ScheduleAgentTask;
+import io.agentscope.extensions.scheduler.config.ScheduleMode;
+import org.quartz.InterruptableJob;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+
+/**
+ * Quartz Job implementation that executes an AgentScope agent task.
+ *
+ * <p>This job retrieves the corresponding {@link QuartzAgentScheduler} and
+ * {@link QuartzScheduleAgentTask} using job data, and then executes the agent's run logic.
+ * It also handles the rescheduling for {@code FIXED_DELAY} tasks.
+ */
+public class AgentQuartzJob implements InterruptableJob {
+
+    private volatile boolean interrupted;
+
+    /**
+     * Executes the agent task.
+     *
+     * @param context The Quartz JobExecutionContext
+     * @throws JobExecutionException if the job execution fails
+     */
+    @Override
+    public void execute(JobExecutionContext context) throws JobExecutionException {
+        if (interrupted) {
+            return;
+        }
+        String schedulerId = context.getJobDetail().getJobDataMap().getString("schedulerId");
+        String taskName = context.getJobDetail().getJobDataMap().getString("taskName");
+        QuartzAgentScheduler scheduler = QuartzAgentSchedulerRegistry.get(schedulerId);
```

Comment:
In cluster mode, this section might leave schedulerId as null, which could lead to task execution skipped. It might be helpful to allow users to specify schedulerId during the build process, with a default fixed string value (e.g., "default-scheduler") as a fallback